instrumentation/opentelemetry_oban/lib/opentelemetry_oban/plugin_handler.ex

```diff
@@ -63,4 +65,53 @@ defmodule OpentelemetryOban.PluginHandler do
 
     OpentelemetryTelemetry.end_telemetry_span(@tracer_id, metadata)
   end
+
+  defp end_span_plugin_attrs(%{plugin: Oban.Plugins.Cron} = metadata) do
+    %{"oban.plugins.cron.jobs_count": length(metadata[:jobs])}
+  end
+
+  defp end_span_plugin_attrs(%{plugin: Oban.Plugins.Gossip} = metadata) do
+    %{"oban.plugins.gossip.gossip_count": metadata[:gossip_count]}
+  end
+
+  defp end_span_plugin_attrs(%{plugin: Oban.Plugins.Lifeline} = metadata) do
+    %{
+      "oban.plugins.lifeline.discarded_count": metadata[:discarded_count],
+      "oban.plugins.lifeline.rescued_count": metadata[:rescued_count]
+    }
+  end
+
+  defp end_span_plugin_attrs(%{plugin: Oban.Plugins.Pruner} = metadata) do
+    %{"oban.plugins.pruner.pruned_count": metadata[:pruned_count]}
+  end
+
+  defp end_span_plugin_attrs(%{plugin: Oban.Pro.Plugins.DynamicCron} = metadata) do
+    %{"oban.pro.plugins.dynamic_cron.jobs_count": length(metadata[:jobs])}
+  end
+
+  defp end_span_plugin_attrs(%{plugin: Oban.Pro.Plugins.DynamicLifeline} = metadata) do
+    %{
+      "oban.pro.plugins.dynamic_lifeline.discarded_count": metadata[:discarded_count],
+      "oban.pro.plugins.dynamic_lifeline.rescued_count": metadata[:rescued_count]
+    }
+  end
+
+  defp end_span_plugin_attrs(%{plugin: Oban.Pro.Plugins.DynamicPrioritizer} = metadata) do
+    %{"oban.pro.plugins.dynamic_prioritizer.reprioritized_count": metadata[:reprioritized_count]}
+  end
+
+  defp end_span_plugin_attrs(%{plugin: Oban.Pro.Plugins.DynamicPruner} = metadata) do
+    %{"oban.pro.plugins.dynamic_pruner.pruned_count": metadata[:pruned_count]}
+  end
+
+  defp end_span_plugin_attrs(%{plugin: Oban.Pro.Plugins.DynamicScaler} = metadata) do
+    %{
+      "oban.pro.plugins.dynamic_scaler.scaler.last_scaled_to": metadata[:scaler][:last_scaled_to],
+      "oban.pro.plugins.dynamic_scaler.scaler.last_scaled_at": metadata[:scaler][:last_scaled_at]
+    }
```

Comment:
It's missing tests

Reply:
I wrote it:
```elixir
  alias OpentelemetryOban.PluginHandler

  describe "handle_plugin_stop/4" do
    test "handle_plugin_stop with Oban.Plugins.Cron" do
      metadata = %{plugin: Oban.Plugins.Cron, jobs: [1, 2, 3]}
      assert PluginHandler.handle_plugin_stop(nil, nil, metadata, nil) == :ok
    end

    test "handle_plugin_stop with Oban.Plugins.Gossip" do
      metadata = %{plugin: Oban.Plugins.Gossip, gossip_count: 5}
      assert PluginHandler.handle_plugin_stop(nil, nil, metadata, nil) == :ok
    end

    test "handle_plugin_stop with Oban.Plugins.Lifeline" do
      metadata = %{plugin: Oban.Plugins.Lifeline, discarded_count: 2, rescued_count: 3}
      assert PluginHandler.handle_plugin_stop(nil, nil, metadata, nil) == :ok
    end

    test "handle_plugin_stop with Oban.Pro.Plugins.DynamicPruner" do
      metadata = %{plugin: Oban.Pro.Plugins.DynamicPruner, pruned_count: 5}
      assert PluginHandler.handle_plugin_stop(nil, nil, metadata, nil) == :ok
    end

    test "handle_plugin_stop with Oban.Pro.Plugins.DynamicScaler" do
      metadata = %{
        plugin: Oban.Pro.Plugins.DynamicScaler,
        scaler: %{last_scaled_to: 10, last_scaled_at: ~U[2022-01-01T00:00:00Z]}
      }

      assert PluginHandler.handle_plugin_stop(nil, nil, metadata, nil) == :ok
    end

    test "handle_plugin_stop with unknown plugin" do
      metadata = %{plugin: UnknownPlugin}
      assert PluginHandler.handle_plugin_stop(nil, nil, metadata, nil) == :ok
    end
  end
```

Reply:
@yordis are we adding these tests?

Reply:
I am working on it; I can not use those tests as is!